src/Provider/AbstractProvider.php

```diff
@@ -323,17 +325,19 @@ protected function fetchUserDetails(AccessToken $token)
     {
         $url = $this->urlUserDetails($token);
 
-        return $this->fetchProviderData($url);
+        $headers = $this->getHeaders($token);
+
+        return $this->fetchProviderData($url, $headers);
     }
 
-    protected function fetchProviderData($url)
+    protected function fetchProviderData($url, array $headers)
```

Comment:
Should we considering initializing this to a default value in case any providers in the wild are calling the `fetchProviderData()` method? Perhaps something as simple as this:

    protected function fetchProviderData($url, array $headers = [])